index.js

```diff
@@ -4,15 +4,155 @@
+  }
+
+  //concurrify :: Applicative m
+  //           => (TypeRep m, () -> m a, (m a, m a) -> m a, (m a, m (a -> b)) -> m b)
+  //           -> Concurrently m
+  return function concurrify(Repr, zero, race, ap){
+
+    if(!isApplicativeRepr(Repr)) invalidArgument('concurrify', 0, 'represent an Applicative', Repr);
+    if(!isFunction(zero)) invalidArgument('concurrify', 1, 'be a function', zero);
+    if(!isFunction(race)) invalidArgument('concurrify', 2, 'be a function', race);
+    if(!isBinary(race)) invalidArgument('concurrify', 2, 'be binary', race);
+    if(!isFunction(ap)) invalidArgument('concurrify', 3, 'be a function', ap);
+    if(!isBinary(ap)) invalidArgument('concurrify', 3, 'be binary', ap);
+
+    var INNERTYPE = getReprType(Repr);
+    var OUTERTYPE = computeType(INNERTYPE);
+
+    function Concurrently(sequential){
+      this.sequential = sequential;
+    }
+
+    function isInner(x){
+      return x instanceof Repr
+      || (Boolean(x) && x.constructor === Repr)
+      || type(x) === Repr[$$type];
+    }
+
+    function isOuter(x){
+      return x instanceof Concurrently
+      || (Boolean(x) && x.constructor === Concurrently)
+      || type(x) === OUTERTYPE;
+    }
+
+    function construct(x){
+      if(!isInner(x)) invalidArgument(OUTERTYPE, 0, 'be of type "' + INNERTYPE + '"', type(x));
+      return new Concurrently(x);
+    }
+
+    var proto = Concurrently.prototype = construct.prototype = {
+      sequential: null,
+      constructor: construct
+    };
+
+    construct[$$type] = OUTERTYPE;
+
+    construct[$zero] = zero;
+
+    construct[$of] = function Concurrently$of(value){
+      return new Concurrently(Z.of(Repr, value));
+    };
+
+    proto[$map] = function Concurrently$map(mapper){
+      if(!isOuter(this)) invalidContext(OUTERTYPE + '#map', this, OUTERTYPE);
+      return new Concurrently(Z.map(mapper, this.sequential));
+    };
+
+    proto[$ap] = function Concurrently$ap(m){
+      if(!isOuter(this)) invalidContext(OUTERTYPE + '#ap', this, OUTERTYPE);
+      if(!isOuter(m)) invalidArgument(OUTERTYPE + '#ap', 0, 'be a Concurrently', m);
+      return new Concurrently(ap(this.sequential, m.sequential));
+    };
+
+    proto[$alt] = function Concurrently$alt(m){
+      if(!isOuter(this)) invalidContext(OUTERTYPE + '#alt', this, OUTERTYPE);
+      if(!isOuter(m)) invalidArgument(OUTERTYPE + '#alt', 0, 'be a Concurrently', m);
+      return new Concurrently(race(this.sequential, m.sequential));
+    };
+
+    proto[$equals] = function Concurrently$equals(m){
+      if(!isOuter(this)) invalidContext(OUTERTYPE + '#equals', this, OUTERTYPE);
+      if(!Z.setoid.test(this.sequential)) invalidContext(OUTERTYPE + '#equals', this, 'Setoid');
+      if(!isOuter(m)) invalidArgument(OUTERTYPE + '#equals', 0, 'be a Concurrently', m);
+      return Z.equals(this.sequential, m.sequential);
+    };
+
+    proto.toString = proto.inspect = function Concurrently$toString(){
```

Comment:
I've assigned it to `inspect` because the same is done in Fluture. I'm aware that theoretically it might blow the stack, but in practice I haven't heard of- nor experienced- it happening. It doesn't sit well with me though, and I think your arguments are convincing. If I change it here I might choose to change it for Fluture as well.